From f4939225cf18a90bc73da2c13af3d6f498c4bb6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 8 Sep 2023 14:24:11 +0200
Subject: [PATCH 0317/1562] winex11: Always ignore MotionNotify event after
 SetCursorPos.

Trying to workaround spurious Wayland mouse motion.

CW-Bug-Id: #22650
---
 dlls/winex11.drv/mouse.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index c052bdd692c..b8bd05f76db 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -1667,7 +1667,7 @@ BOOL X11DRV_MotionNotify( HWND hwnd, XEvent *xev )
     input.mi.time        = x11drv_time_to_ticks( event->time );
     input.mi.dwExtraInfo = 0;
 
-    if (!hwnd && is_old_motion_event( event->serial ))
+    if (is_old_motion_event( event->serial ))
     {
         TRACE( "pos %d,%d old serial %lu, ignoring\n", event->x, event->y, event->serial );
         return FALSE;
#@@ -1929,7 +1929,7 @@ static BOOL X11DRV_XIDeviceEvent( XIDeviceEvent *event )
#     TRACE( "evtype %u hwnd %p/%lx pos %f,%f detail %u flags %#x serial %lu\n",
#            event->evtype, hwnd, event->event, event->event_x, event->event_y, event->detail, event->flags, event->serial );
# 
#-    if (!hwnd && is_old_motion_event( event->serial ))
#+    if (is_old_motion_event( event->serial ))
#     {
#         TRACE( "pos %f,%f old serial %lu, ignoring\n", event->event_x, event->event_y, event->serial );
#         return FALSE;
-- 
2.45.2

